tests/integration/modifier-managers/oo-modifiers-classic-test.js

```diff
@@ -193,51 +393,107 @@ module('Integration | Modifier Manager | oo modifier (classic)', function(hooks)
-    ]);
+      this.set('value', 1);
+      await settled();
+      this.set('shouldRender', false);
+      await settled();
+      assert.deepEqual(callstack, [
+        'didInsertElement',
+        'didReceiveArguments',
+        'didReceiveArguments',
+        'didUpdateArguments',
+        'willDestroyElement'
+      ]);
+    });
+
+    test('can participate in ember dependency injection', async function(assert) {
+      this.owner.register(
+        'service:test-service',
+        Service.extend({ value: 'test-service-value' })
+      );
+      this.registerModifierClass(
+        'songbird',
+        Modifier.modifier(
+          Modifier.extend({
+            testService: service(),
+            didInsertElement() {
+              assert.equal(this.testService.value, 'test-service-value');
+            }
+          })
+        )
+      );
+      await render(hbs`<h1 {{songbird}}>Hello</h1>`);
+    });
   });
 
-  test('can participate in ember dependency injection', async function(assert) {
-    this.owner.register(
-      'service:test-service',
-      Service.extend({ value: 'test-service-value' })
-    );
-    this.registerModifierClass(
-      'songbird',
-      Modifier.extend({
-        testService: service(),
-        didInsertElement() {
-          assert.equal(this.testService.value, 'test-service-value');
-        }
-      })
-    );
-    await render(hbs`<h1 {{songbird}}>Hello</h1>`);
+  module('Lifecycle and dependency without calling Modifier.modifier', function() {
+    test('has correct lifecycle hooks ordering', async function(assert) {
+      let callstack = [];
+      this.value = 0;
+      this.shouldRender = true;
+      this.registerModifierClass(
+        'songbird',
+        Modifier.extend({
+          didInsertElement() { callstack.push('didInsertElement'); },
+          didReceiveArguments() { callstack.push('didReceiveArguments'); },
+          didUpdateArguments() { callstack.push('didUpdateArguments'); },
+          willDestroyElement() { callstack.push('willDestroyElement'); }
+        })
+      );
+      await render(hbs`
+      {{#if this.shouldRender}}
+        <h1 {{songbird this.value}}>Hey</h1>
+      {{/if}}
+    `);
+      this.set('value', 1);
+      await settled();
+      this.set('shouldRender', false);
+      await settled();
+      assert.deepEqual(callstack, [
+        'didInsertElement',
+        'didReceiveArguments',
+        'didReceiveArguments',
+        'didUpdateArguments',
+        'willDestroyElement'
+      ]);
+    });
+
+    test('can participate in ember dependency injection', async function(assert) {
+      this.owner.register(
+        'service:test-service',
+        Service.extend({ value: 'test-service-value' })
+      );
+      this.registerModifierClass(
+        'songbird',
+        Modifier.extend({
+          testService: service(),
+          didInsertElement() {
+            assert.equal(this.testService.value, 'test-service-value');
+          }
+        })
```

Comment:
Should we be testing that `Modifier.modifier` has been deprecated?

Reply:
I've asked how to do this in Ember Discord#topic-testing.  If there's a reasonable response on how to test this, I'll add that test, otherwise I'll press that merge button.